cli/cmd/install.go

```diff
@@ -1108,3 +1216,18 @@ func toIdentityContext(idvals *identityWithAnchorsAndTrustDomain) *pb.IdentityCo
 		Scheme:             idvals.Identity.Issuer.Scheme,
 	}
 }
+func checkAddon(values *l5dcharts.Values, name string) (addonvalues []byte, enabled bool) {
+
+	r := reflect.ValueOf(values)
+
+	if !reflect.Indirect(r).FieldByName(strings.Title(name)).IsNil() {
+		if reflect.Indirect(r).FieldByName(strings.Title(name)).MapIndex(reflect.ValueOf("enabled")).Interface().(bool) {
+			values, err := yaml.Marshal(reflect.Indirect(r).FieldByName(strings.Title(name)).Interface())
+			if err != nil {
+				return nil, false
+			}
+			return values, true
+		}
```

Comment:
if the addon config is missing then the value of `reflect.Indirect(r).FieldByName(strings.Title(name))` would be an empty map (zero value) to which you can safely apply `MapIndex`, so no need of the nested `if` nor the `IsNil()` check. 

You can also extract the common part of the remaining condition:
```go
		config := reflect.Indirect(r).FieldByName(strings.Title(name)); 
                if config.MapIndex(reflect.ValueOf("enabled")).Interface().(bool) {
			values, err := yaml.Marshal(config.Interface())
			if err != nil {
				return nil, false
			}
			return values, true
		}
```